src/shared/components/common/error-guard.tsx

```diff
@@ -0,0 +1,26 @@
+import { Component } from "inferno";
+import { ErrorPageData, setIsoData } from "../../utils";
+import { ErrorPage } from "../app/error-page";
+
+class ErrorGuard extends Component<any, any> {
```

Comment:
I just thought of something. I could probably handle this on the server. It would probably both reduce bundle size and make my hacky use of 'routeData' unnecessary.

Reply:
I just know that the server will return the json `{"error: "string"}`, for any failed request. Not sure how that translates for this. 

In jerboa at least, all these errors do proper exception throws, so I can catch them, read the error message, and do actions based on that.

Reply:
My comment was unclear. I meant on the server side of the UI app that serves the pages, but not the actual Lemmy server.

As to doing different actions based on different error messages, that's definitely doable. We just need to defined what should be done for those error messages. At the very least there should be a way to handle displaying that a user is banned (I even think there's an open issue for that).